python/ray/autoscaler/launch_and_verify_cluster.py

```diff
@@ -45,12 +43,68 @@ def check_arguments():
         default=3,
         help="Number of retries for verifying Ray is running (default: 3)",
     )
+    parser.add_argument(
+        "--num-expected-nodes",
+        type=int,
+        default=1,
+        help="Number of nodes for verifying Ray is running (default: 1)",
+    )
+    parser.add_argument(
+        "--docker-override",
+        choices=["disable", "latest", "nightly", "commit"],
+        default="disable",
+        help="Override the docker image used for the head node and worker nodes",
+    )
+    parser.add_argument(
+        "--wheel-override",
+        type=str,
+        default="",
+        help="Override the wheel used for the head node and worker nodes",
+    )
     parser.add_argument(
         "cluster_config", type=str, help="Path to the cluster configuration file"
     )
     args = parser.parse_args()
 
-    return args.cluster_config, args.retries, args.no_config_cache
+    assert not (
+        args.docker_override != "disable" and args.wheel_override != ""
+    ), "Cannot override both docker and wheel"
+
+    return (
+        args.cluster_config,
+        args.retries,
+        args.no_config_cache,
+        args.num_expected_nodes,
+        args.docker_override,
+        args.wheel_override,
+    )
+
+
+def get_docker_image(docker_override):
+    """
+    Get the docker image to use for the head node and worker nodes.
+
+    Args:
+        docker_override: The value of the --docker-override flag.
+
+    Returns:
+        The docker image to use for the head node and worker nodes, or None if not
+        applicable.
+    """
+    if docker_override == "latest":
+        return "rayproject/ray:latest-py38"
+    elif docker_override == "nightly":
+        return "rayproject/ray:nightly-py38"
+    elif docker_override == "commit":
+        if re.match("^[0-9]+.[0-9]+.[0-9]+$", ray.__version__):
+            return f"rayproject/ray:{ray.__version__}.{ray.__commit__[:6]}-py38"
```

Comment:
so I think this will be using the driver script's ray version? I guess it will be more flexible if we could pass in a commit (or just a docker image) in?